File: java/src/br/unifor/facilitador/porta/AgendaService.java
package br.unifor.facilitador.porta;

import br.unifor.facilitador.dominio.*;
import java.time.LocalDateTime;
import java.util.UUID;

/** Ciclo de vida do agendamento. Os tres metodos giram em torno do mesmo objeto. */
public interface AgendaService {
    Consulta agendar(Paciente p, Medico m, LocalDateTime dt, Modalidade modalidade);
    void cancelar(UUID id, String motivo);
    void remarcar(UUID id, LocalDateTime nova);
}

File: src/AgendaService.ts
import Paciente from "./Paciente.js";
import Medico from "./Medico.js";
import Consulta from "./Consulta.js";
import { UUID, DateTime } from "./Tipos.js";

export default interface AgendaService {
  agendar(p: Paciente, m: Medico, dt: DateTime): Consulta;
  cancelar(id: UUID, motivo: string): void;
  remarcar(id: UUID, nova: DateTime): void;
}
